### ISSUE TITLE
CTRL-drag causes crash

### ISSUE THREAD

**Issue by SKYMEMiC** [pyfa-org/Pyfa#1186]
Tried to ctrl-drag the drone damage amplifier as this error appeared
-----
```
OS version: 	Windows-8-6.2.9200
Python: 	2.7.10
wxPython: 	3.0.2.0
SQLAlchemy: 	1.0.5
Logbook: 	1.0.0
pyfa version: 1.29.2 Stable - YC119.5 1.0
pyfa root: C:\Program Files (x86)\pyfa
save path: C:\Users\user\.pyfa
fs encoding: mbcs

EXCEPTION: 'NoneType' object is not iterable

  File "C:\Program Files (x86)\pyfa\library.zip\gui\builtinViews\fittingView.py", line 115, in OnData
    self.dropFn(x, y, data)
  File "C:\Program Files (x86)\pyfa\library.zip\gui\builtinViews\fittingView.py", line 201, in handleListDrag
    self.swapItems(x, y, int(data[1]))
  File "C:\Program Files (x86)\pyfa\library.zip\gui\builtinViews\fittingView.py", line 426, in swapItems
    sFit.cloneModule(self.mainFrame.getActiveFit(), srcIdx, mod2.modPosition)
  File "C:\Program Files (x86)\pyfa\library.zip\service\fit.py", line 685, in cloneModule
    if new.fits(fit):
  File "C:\Program Files (x86)\pyfa\library.zip\eos\saveddata\module.py", line 401, in fits
    fits = self.__fitRestrictions(fit, hardpointLimit)
  File "C:\Program Files (x86)\pyfa\library.zip\eos\saveddata\module.py", line 418, in __fitRestrictions
    for attr in self.itemModifiedAttributes.keys():
  File "c:\python\32-bit\2.7\lib\_abcoll.py", line 410, in keys
  File "C:\Program Files (x86)\pyfa\library.zip\eos\modifiedAttributeDict.py", line 152, in __iter__
    all_dict = dict(self.original, **self.__modified)
```
----
Fit:

```
[Stratios, Explorer]

Damage Control II
[Empty Low slot]
[Empty Low slot]
Drone Damage Amplifier II
Drone Damage Amplifier II

Adaptive Invulnerability Field II
X-Large Ancillary Shield Booster, Navy Cap Booster 800
Medium Capacitor Booster II, Navy Cap Booster 800
Data Analyzer I
100MN Y-S8 Compact Afterburner

[Empty High slot]
[Empty High slot]
[Empty High slot]
Covert Ops Cloaking Device II
Sisters Core Probe Launcher, Sisters Core Scanner Probe

Medium Processor Overclocking Unit I
Medium Processor Overclocking Unit I
[Empty Rig slot]


Republic Fleet Berserker x4
Republic Fleet Valkyrie x5
Imperial Navy Praetor x5
Imperial Navy Infiltrator x5
Republic Fleet Warrior x5
Imperial Navy Acolyte x5
Hornet EC-300 x5
```

**Comment by blitzmann** [comment]
Hi there,

Can you be a bit more specific with where you're dragging this to? 

![image](https://cloud.githubusercontent.com/assets/3904767/26281462/356bfe60-3dc5-11e7-9705-5d4e43a5ae78.png)

I attempted to copy one of the drone amplifiers to the first empty low slot (4th overall), and could not reproduce the error. I tried other combinations as well with no luck.

What did produce an error, however, was trying to clone an empty slot to another empty slot. It produces the exact same error. Could this be what you did? Can you reproduce consistently? :)

**Comment by blitzmann** [comment]
@SKYMEMiC can you test and see if this build works better for you?

https://ci.appveyor.com/project/blitzmann/pyfa/build/bug/1186-101/artifacts

I found a bunch of issue with the swapping of modules from various sources - I believe I've cleaned them most if not all of them up. If you still run into the error, please let me know exactly which module you're dragging, where it's dragging from, and where it's dragging to (the slots). include pictures if it's easier :)

**Comment by SKYMEMiC** [comment]
Sorry for not being more precise, but i didn't know how to reproduce. I just noticed before some strange behavior and as the bug window came i just wanted to let you know. I installed the new build. Will test it. As I have the chance to change something: I often use ctrl-drag to copy modules. Often it happens to me, that a module is deleted than copied. I think it's the "delete on double click"-function. I select the module and use to press ctrl before moving a module. Then selecting the new copy, pressing ctrl and copying by dragging again. The "delete on double click"-function executes of being too fast and is not taking notice of a pressed ctrl-key.

**Comment by blitzmann** [comment]
@SKYMEMiC I see what you mean, however I don't know of a good way to prevent that. I checked for a double click up event which would only fire after the button is released after the second click, however wxpython doesn't seem to support that (it may very well be a system thing). The best solution I can come up with is to ignore the `removeModule` call if the ctrl key is pressed, however in my testing it can still be very problematic (since often times, at least for me, the double click event registers before I can actually press the ctrol key). 

Double click to remove module is so engrained into the user experience of the program that removing it is not an option - way too many people would be upset lol. I can only suggest that you keep mindful of the speed at which you click. You may also want to go into your system settings and tweak the delay for double click on your mouse.

Anyway, let me know if the issue you brought up seems to be fixed in that build. If it is, I'll mark this as fixed :)

**Comment by SKYMEMiC** [comment]
What about an entry in Preferences -> General: (x) Double click causes deletion

**Comment by blitzmann** [comment]
> What about an entry in Preferences -> General: (x) Double click causes deletion

That's always possible, but seems to be very specific for this case. I'll continue to look for possible solutions in handling the event :)

Were you ever able to test out that build for the original issue? Is that working for you?

**Comment by SKYMEMiC** [comment]
I've been testing around quite a bit and no further error appeared to me. But I don't know whether it's just a feeling or really true that Pyfa now takes longer to start. (Are there debuging Options enabled that could cause that?)

**Comment by blitzmann** [comment]
@SKYMEMiC 

> But I don't know whether it's just a feeling or really true that Pyfa now takes longer to start. (Are there debuging Options enabled that could cause that?)

There aren't any debugging options on those automated builds that I'm aware of. I tested it just now and the initial load indeed took a little while to start, but subsequent loads seem to be the same as with my personal installation of pyfa. It could be an anomoly related to the fact that these are automated builds. v0v

**Comment by blitzmann** [comment]
Marking this as fixed since I've merged in #1187 and #1190, which should fix the random errors on drag and drop and adds the ignore if CTRL is pressed. :)
